diff --git a/lib/cjs/useWallets.js b/lib/cjs/useWallets.js
index b2281ea..78d15ee 100644
--- a/lib/cjs/useWallets.js
+++ b/lib/cjs/useWallets.js
@@ -9,7 +9,7 @@ const useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT_js_1 = require("./useWallets_INTER
  */
 function useWallets() {
     const wallets = (0, useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT_js_1.useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT)();
-    const uiWallets = (0, react_1.useMemo)(() => wallets.map(ui_registry_1.getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED), [wallets]);
+    const uiWallets = (0, react_1.useMemo)(() => wallets.filter(Boolean).map(ui_registry_1.getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED), [wallets]);
     return uiWallets;
 }
 //# sourceMappingURL=useWallets.js.map
\ No newline at end of file
diff --git a/lib/esm/useWallets.js b/lib/esm/useWallets.js
index 07376ed..ed57aad 100644
--- a/lib/esm/useWallets.js
+++ b/lib/esm/useWallets.js
@@ -6,7 +6,7 @@ import { useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT } from './useWallets_INTERNAL_O
  */
 export function useWallets() {
     const wallets = useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT();
-    const uiWallets = useMemo(() => wallets.map(getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED), [wallets]);
+    const uiWallets = useMemo(() => wallets.filter(Boolean).map(getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED), [wallets]);
     return uiWallets;
 }
 //# sourceMappingURL=useWallets.js.map
\ No newline at end of file
diff --git a/src/useWallets.ts b/src/useWallets.ts
index 91e48df..226257d 100644
--- a/src/useWallets.ts
+++ b/src/useWallets.ts
@@ -10,7 +10,10 @@ import { useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT } from './useWallets_INTERNAL_O
 export function useWallets(): readonly UiWallet[] {
     const wallets = useWallets_INTERNAL_ONLY_NOT_FOR_EXPORT();
     const uiWallets = useMemo(
-        () => wallets.map(getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED),
+        () =>
+            wallets
+                .filter((wallet): wallet is NonNullable<typeof wallet> => Boolean(wallet))
+                .map(getOrCreateUiWalletForStandardWallet_DO_NOT_USE_OR_YOU_WILL_BE_FIRED),
         [wallets]
     );
     return uiWallets;
